docs/tutorial/agent-on-docker.md

```diff
@@ -0,0 +1,155 @@
+However, there may be cases who want to use only Vald Agent without kubernetes.
+
+This article will show you how to deploy and run the Vald Agent on Docker.
+Fashion-mnist is used as an example of a dataset as same as [Get Stared](../tutorial/get-started.md)
+
+## Requirements
+
+- docker: v19.0 ~
+- go: v1.14 ~
+- libhdf5 (_only required for this tutorial._)
+
+
+Hdf5 is required for this tutorial. If hdf5 is not installed, please install [hdf5](https://www.hdfgroup.org/).
+<details><summary>[Optional] Install hdf5</summary><br>
+
+```bash
+# yum
+yum install -y hdf5-devel
+
+# apt
+apt-get install libhdf5-serial-dev
+
+# homebrew
+brew install hdf5
+```
+</details>
+
+## Deploy
+
+This chapter will show you how to deploy Vald Agent on docker.<br>
+This chapter will use NGT for the core method of Vald Agent.
+
+1. Clone the vdaas/vald repository
+
+    ```bash
+    git clone https://github.com/vdaas/vald.git
+    ```
+
+1. Create `config.yaml`
+
+    The configuration of Vald agent for docker is set using `config.yaml`<br>
+    You also check [the sample](https://github.com/vdaas/vald/blob/master/cmd/agent/core/ngt/sample.yaml).
+
+    ```bash
+    ---
+    version: v0.0.0
+    time_zone: JST
+    logging:
+      logger: glg
+      level: debug
+      format: json
+    server_config:
+      servers:
+        - name: agent-grpc
+          host: 0.0.0.0
+          port: 8081
+          mode: GRPC
+          probe_wait_time: "3s"
+          http:
+            shutdown_duration: "5s"
+            handler_timeout: ""
+            idle_timeout: ""
+            read_header_timeout: ""
+            read_timeout: ""
+            write_timeout: ""
+      startup_strategy:
+        - agent-grpc
+      shutdown_strategy:
+        - agent-grpc
+      full_shutdown_duration: 600s
+      tls:
+        enabled: false
+        # cert: /path/to/cert
+        # key: /path/to/key
+        # ca: /path/to/ca
+    ngt:
+      # vector dimension
+      dimension: 784
+      # bulk insert chunk size
+      bulk_insert_chunk_size: 10
+      # distance_type, which should be `l1`, `l2`, `angle`, `hamming`, `cosine`, `normalizedangle` or `nomralizedcosine`
+      distance_type: l2
+      # object_type, which should be `float` or `uint8`
+      object_type: float
+      # creation edge size
+      creation_edge_size: 20
+      # search edge size
+      search_edge_size: 10
+      # in-memory mode enabled
+      enable_in_memory_mode: true
+      # The limit duration of automatic indexing 
+      auto_index_duration_limit: 1m
+      # Check duration of automatic indexing.
+      auto_index_check_duration: 10s
+      # The number of cache to trigger automatic indexing
+      auto_index_length: 100
+    ```
+
+1. Deploy Vald Agent on Docker
+
+    To deploy Vald agent on docker with `config.yaml`, you can run below command.
+
+    ```bash
+    docker run -v path-to-dir-of-config.yaml:/etc/server -p 8081:8081 --rm --it vdaas/vald-agent-ngt
```

Comment:
this command failed...
```
% docker run -v ./config.yaml:/etc/server -p 8081:8081 --rm --it vdaas/vald-agent-ngt
unknown flag: --it
See 'docker run --help'.
```

Reply:
`docker run -it` is correct 😅

Reply:
🤔
```
 % docker run --it -v ./config.yaml:/etc/server -p 8081:8081 --rm vdaas/vald-agent-ngt
unknown flag: --it
See 'docker run --help'.
wdiu@MBP13UAQ-A070 vald % docker version
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

Reply:
seems it is not working... and no output for this command.
```
 % docker run -it -v $(pwd)/config.yaml:/etc/server/config.yaml -p 8081:8081 --rm vdaas/vald-agent-ngt

```

Reply:
To mount config.yaml, you should set parent directory of `config.yaml` like a below
`docker run -p 8081:8081 --rm -v ${pwd):/etc/server vdaas/vald-agent-ngt`

Could you try the above command?

Reply:
it should be `$(pwd)` instead of `${pwd)`. but i suggest to specify the file path instead of folder path which works for me also. ` docker run -p 8081:8081 --rm -v $(pwd)/config.yaml:/etc/server/config.yaml -it vdaas/vald-agent-ngt`
```bash
% docker run -p 8081:8081 --rm -v $(pwd):/etc/server -it vdaas/vald-agent-ngt
{"Date":"2020-06-29T05:21:37.897641Z","Level":"INFO","Detail":"maxprocs: Leaving GOMAXPROCS=2: CPU quota undefined"}
{"Date":"2020-06-29T05:21:37.897641Z","Level":"INFO","Detail":"service agent ngt v0.0.0 starting..."}
{"Date":"2020-06-29T05:21:37.897641Z","Level":"INFO","Detail":["daemon start"]}
{"Date":"2020-06-29T05:21:37.897641Z","Level":"INFO","Detail":"server agent-grpc executing preStartFunc"}
{"Date":"2020-06-29T05:21:37.897641Z","Level":"INFO","Detail":"gRPC server agent-grpc starting on 0.0.0.0:8081"}
```